server/pulp/server/content/web/views.py

```diff
@@ -131,13 +131,14 @@ def get(self, request):
             # Not Authorized
             return HttpResponseForbidden()
 
-        # Already downloaded
-        if os.path.exists(path):
-            return self.x_send(path)
-
-        # Redirect
-        if lazy_enabled:
-            return self.redirect(request, self.key)
+        if not os.path.isdir(path):
+            # Already downloaded and is a file
+            if os.path.exists(path):
+                return self.x_send(path)
+
+            # Redirect if lazy is on and the link exists in the repo
+            if lazy_enabled and os.path.lexists(request.path_info):
```

Comment:
`request.path_info` is the path the client requested (`/var/www/pub/yum/https/repos/.../some_symlink.rpm`) whereas `path` is the file that symlink points to.